Sources/Optimizely+Decide/OptimizelyUserContext.swift

```diff
@@ -122,15 +135,105 @@ public class OptimizelyUserContext {
         guard let optimizely = self.optimizely, optimizely.config != nil else {
             throw OptimizelyError.sdkNotReady
         }
-
+        
         try optimizely.track(eventKey: eventKey,
                              userId: userId,
                              attributes: attributes,
                              eventTags: eventTags)
     }
+    
+}
 
+// MARK: - ForcedDecisions
+
+extension OptimizelyUserContext {
+    
+    public func setForcedDecision(flagKey: String,
+                                  ruleKey: String? = nil,
+                                  variationKey: String) -> Bool {
+        
+        guard let optimizely = self.optimizely, optimizely.config != nil else {
+            logger.e(OptimizelyError.sdkNotReady)
+            return false
+        }
+        
+        if let index = savedDecisions.firstIndex(where: { $0.flagKey == flagKey && $0.ruleKey == ruleKey }) {
+            savedDecisions[index].variationKey = variationKey
+        } else {
+            savedDecisions.append(SavedDecision(flagKey: flagKey, ruleKey: ruleKey, variationKey: variationKey))
+        }
+        
+        return true
+    }
+    
+    public func getForcedDecision(flagKey: String, ruleKey: String? = nil) -> String? {
+        guard let optimizely = self.optimizely, optimizely.config != nil else {
+            logger.e(OptimizelyError.sdkNotReady)
+            return nil
+        }
+        
+        return findForcedDecision(flagKey: flagKey, ruleKey: ruleKey)
+    }
+    
+    public func removeForcedDecision(flagKey: String, ruleKey: String? = nil) -> Bool {
+        guard let optimizely = self.optimizely, optimizely.config != nil else {
+            logger.e(OptimizelyError.sdkNotReady)
+            return false
+        }
+        
+        if let index = savedDecisions.firstIndex(where: { $0.flagKey == flagKey && $0.ruleKey == ruleKey }) {
+            _ = savedDecisions.remove(at: index)
+            return true
+        }
+        
+        return false
+    }
+    
+    public func removeForcedDecisions() -> Bool {
+        guard let optimizely = self.optimizely, optimizely.config != nil else {
+            logger.e(OptimizelyError.sdkNotReady)
+            return false
+        }
+        
+        savedDecisions.removeAll()
+        return true
+    }
+    
+    func findForcedDecision(flagKey: String, ruleKey: String? = nil) -> String? {
```

Comment:
this method should return reason.

Reply:
This is for a simple lookup. The reasons are included in "findValidatedForcedDecision" with the result of "findForcedDecision".